### ISSUE TITLE
Feature Request: When adding a custom domain make bind certificate optional (like in App Service)

### ISSUE THREAD

**Issue by johnmccormick99** [microsoft/azure-container-apps#665]
**Is your feature request related to a problem? Please describe**.  
When adding a custom domain to Azure Container App you are forced to bind a certificate. This is problematic if you are using a Front Door custom domain managed certificate as it cannot be exported.

**Describe the solution you'd like**.    
Be able to add a custom domain to Azure Container App without having to bind a certificate. You can do this in App Service https://vincentlauzon.com/2019/07/17/azure-app-service-authentication-with-azure-front-door/

**Describe alternatives you've considered**.   
We will have to use bring-your-own certificate on Front Door. We don't want to manage certificates on Front Door and Azure Container App.

**Additional context**.  
The scenario of fronting web apps with Front Door is very common and the case for moving to Azure Container App weakens when its capabilities do not match App Service.

I hope that I've missed a trick and it in fact can be done (perhaps using Bicep) and it's just the Azure Portal that imposes these restrictions.

Many thanks,

**Comment by anthonychu** [comment]
If you use a container app as a backend in Front Door, you shouldn't need to use a custom domain in the container app. Are you able to use the default FQDN provided by Container Apps for this?

**Comment by johnmccormick99** [comment]
Hi. 

That works fine. It doesn't work if you enable Azure AD authentication on the Container App as the host header set in the Front Door backend is used by Container App to build the authorization redirect URL. It will redirect the user to Container App and bypass Front Door.

If you want the authorization redirect URL to be Front Door you must set the host header to be Front Door. This requires a custom domain to be added to Container App to match Front Door. But to add the custom domain to Container App you must bind a certificate, which means you cannot take advantage of the managed certs on Front Door.

There's a similar challenge with App Service but it's achievable as you can add a custom domain to App Service without having to bind a certificate as discussed in this post https://vincentlauzon.com/2019/07/17/azure-app-service-authentication-with-azure-front-door/.

So in summary, we want to use Front Door with a custom domain and managed certificate + Container App with Easy Auth (Azure AD). We can do that if we can add a custom domain to container app without a cert.

Many thanks.

**Comment by SophCarp** [comment]
a work item for our team has been created ( # 17547371)

**Comment by danielmatthews84** [comment]
Will this be added to the roadmap at all?

It's a real pain to still have to bind a cert to a container app when off loading TLS in App Gateway. Especially given that ACA can't resolve certs from KeyVault yet.

**Comment by jankenr** [comment]
Still very much needed since we are using Cloudflare for managing DNS, certificates and rules and use ACA to host the application instead of App Service which is a bit outdated

**Comment by dextry92** [comment]
Hello all, 

Just wanted to provide an update regarding the internal work item 17547371. This has been completed and now the platform supports binding custom domain with no cert for container apps. 

Thank you!

**Comment by dschmittwilken** [comment]
I configured a custom domain (without certificate "No binding") on the azure container app, but still the auth redirect goes to the internal domain. Any ideas? I have a frontdoor with same custom domain verified using AFD Managed Certificate.
